src/conflictresolutionscreen.cpp

```diff
@@ -0,0 +1,138 @@
+ ************************************************************************************
+ *  This program's developed as part of GSoC - 2016
+ *  Project: One Click Installer
+ *  Mentors: Antonio Larrosa, and Cornelius Schumacher
+ *  Organization: OpenSUSE
+ *  Previous Contributor: None
+ ***********************************************************************************/
+
+#include <QApplication>
+#include <QDBusConnection>
+#include <QRadioButton>
+#include "conflictresolutionscreen.h"
+#include "oneclickinstalleradaptor.h"
+
+ConflictResolutionScreen::ConflictResolutionScreen()
+{
+    setStyleSheet( "background-color: white;" );
+    
+    new OneclickinstallerAdaptor( this );
+    QDBusConnection connection = QDBusConnection::systemBus();
+    if ( !connection.isConnected() ) {
+	qFatal( "Cannot connect to the D-Bus system bus" );
+	exit( 1 );
+    }
+    connection.registerObject( "/OCI", this );
+    if ( !connection.registerService("org.opensuse.oneclickinstaller") ) {
+	qDebug() << qPrintable( QDBusConnection::systemBus().lastError().message() );
+        exit( 1 );
+    }
+   
+    m_mainLayout = new QVBoxLayout;
+    
+    //QObject::connect( m_ociHelper, SIGNAL( displayProblemAndSolutions( QString, QString ) ), this, SLOT( problemSolutionWidget( QString, QString ) ) );
+    QDBusConnection sysBus = QDBusConnection::systemBus();
+    if ( !sysBus.isConnected() ) {
+	qFatal( "Cannot connect to the D-Bus system bus" );
+	exit( 1 );
+    }
+    sysBus.connect( QString(), QString(), "org.opensuse.OCIHelper", "displayProblemAndSolutions", this, SLOT( problemSolutionWidget( QString, QString ) ) );
+   
+    setLayout ( m_mainLayout );
+}
+
+void ConflictResolutionScreen::problemSolutionWidget( QString probDescription, QString solutions )
+{
+    m_solId = -1;
+    if ( m_solutionWidget != NULL && m_questionLabel != NULL ) {
+	m_mainLayout->removeWidget( m_questionLabel );
+	m_mainLayout->removeWidget( m_solutionWidget );
+    }
+    QLabel *questionLabel = new QLabel( probDescription );
+    questionLabel->setStyleSheet( "background-color: rgb(254, 250, 210); border-bottom : 1px solid rgb(252,233,79); border-left : 1px solid rgb(196,181,147); border-top : 1px solid rgb(196,181,147); border-right : 1px solid rgb(196,181,147);" );
+    questionLabel->setFixedHeight( 50 );
+    questionLabel->setWordWrap( true );
+    
+    QWidget *solutionWidget = new QWidget;
+    solutionWidget->setObjectName( "solutionWidget" );
+    solutionWidget->setStyleSheet( "QWidget#solutionWidget{ background-color : white; border-bottom: 1px solid rgb(196, 181, 147); border-left : 1px solid rgb(196,181,147); border-top : 1px solid rgb(196,181,147); border-right : 1px solid rgb(196,181,147); }" );
+    
+    QVBoxLayout *solutionLayout = new QVBoxLayout( solutionWidget );
+    solutionLayout->setSpacing( 0 );
+    
+    for ( QString solution : solutions.split( "$", QString::SkipEmptyParts) ) {
+	QRadioButton *sol = new QRadioButton( solution );
```

Comment:
You can add a QButtonGroup to the class and simplify the QRadioButton code a bit. You'll have to create a QButtonGroup before and after creating QRadioButton *sol as you did, do:
m_buttonGroup.addButton( sol, radioID++ );
(set "int radioID=0;" before the for loop)